devices/HumanDataCollector/HumanDataCollector.cpp

```diff
@@ -0,0 +1,663 @@
+    std::vector<double> jointTorques;
+
+    // Yarp ports for streaming data from IHumanState interface of HumanStateProvider
+    yarp::os::BufferedPort<yarp::sig::Vector> basePoseDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> baseVelocityDataPort;
+    yarp::os::BufferedPort<yarp::os::Bottle> stateJointNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> jointPositionsDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> jointVelocitiesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comPositionDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comVelocityDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comProperAccelerationInBaseFrameDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comProperAccelerationInWorldFrameDataPort;
+
+    // Yarp ports for streaming data from IHumanWrench interface of HumanWrenchProvider
+    yarp::os::BufferedPort<yarp::os::Bottle> wrenchMeasurementSourceNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> wrenchMeasurementValuesDataPort;
+
+    // Yarp ports for streaming data from IHumanWrench interfce and IHumanDynamics interface of HumanDynamicsEstimator
+    yarp::os::BufferedPort<yarp::os::Bottle> wrenchEstimateSourceNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> wrenchEstimateValuesDataPort;
+
+    yarp::os::BufferedPort<yarp::os::Bottle> dynamicsJointNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> jointTorquesDataPort;
+};
+
+HumanDataCollector::HumanDataCollector()
+    : PeriodicThread(DefaultPeriod)
+    , pImpl{new impl()}
+{}
+
+HumanDataCollector::~HumanDataCollector() {}
+
+bool HumanDataCollector::open(yarp::os::Searchable &config) {
+
+    // ===============================
+    // CHECK THE CONFIGURATION OPTIONS
+    // ===============================
+
+    if (!(config.check("period") && config.find("period").isFloat64())) {
+        yInfo() << LogPrefix << "Using default period";
+    }
+
+    if (!(config.check("portPrefix") && config.find("portPrefix").isString())) {
+        yInfo() << LogPrefix << "Using default portPrefix HDE ";
+    }
+
+    // ===============================
+    // PARSE THE CONFIGURATION OPTIONS
+    // ===============================
+
+    double period = config.check("period", yarp::os::Value(0.01)).asFloat64();
+    pImpl->portPrefix = config.check("portPrefix", yarp::os::Value("HDE")).asString();
+
+    //TODO: Define rpc to reset the data dump or restarting the visualization??
+
+    yInfo() << LogPrefix << "*** ===========================";
+    yInfo() << LogPrefix << "*** Period                    :" << period;
+    yInfo() << LogPrefix << "*** portPrefix                :" << pImpl->portPrefix;
+    yInfo() << LogPrefix << "*** ===========================";
+
+    // Set period
+    setPeriod(pImpl->period);
+
+    return true;
+}
+
+bool HumanDataCollector::close() {
+    return true;
+}
+
+void HumanDataCollector::threadRelease() {}
```

Comment:
I have no idea what this is. Do we need it?

Reply:
@diegoferigo Good call, seems redundant https://www.yarp.it/classyarp_1_1os_1_1PeriodicThread.html#af8bb72b93be3823a3d4e217f959a612f, as we handle the cleanup in `detach()`